README.md

```diff
@@ -233,13 +233,154 @@ mstorage.info("<media_id>", new CompletionHandler<MediaInfo>() {
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+### Get specific media metadata
+
+* Exif
+
+```java
+mstorage.exif("<media_id>", new CompletionHandler<Map<String, String>>() {
+    @Override
+    public void onCompleted(Map<String, String> exif) {
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+* Google Photo Sphere XMP
+
+```java
+mstorage.gpano("<media_id>", new CompletionHandler<Map<String, String>>() {
+    @Override
+    public void onCompleted(Map<String, String> gpano) {
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+* All user metadata
+
+```java
+mstorage.userMeta("<media_id>", new CompletionHandler<Map<String, String>>() {
+    @Override
+    public void onCompleted(Map<String, String> userMeta) {
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+* Specific user metadata
+
+```java
+mstorage.userMeta("<media_id>", "<key>", new CompletionHandler<String>() {
+    @Override
+    public void onCompleted(String value) {
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+### Attach user metadata to media
+
+```java
+mstorage.addUserMeta("<media_id>", "<key>", "<value>", new CompletionHandler<Object>() {
+    @Override
+    public void onCompleted(Object result) {
+        // Do something.
+        // result may be empty.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+### Delete specified user metadata from media
+
+```java
+mstorage.deleteUserMeta("<media_id>", "<key>", new CompletionHandler<Object>() {
+    @Override
+    public void onCompleted(Object result) {
+        // Do something.
+        // result may be empty.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+### Delete all user metadata from media
+
+```java
+mstorage.deleteAllUserMeta("<media_id>", new CompletionHandler<Object>() {
+    @Override
+    public void onCompleted(Object result) {
+        // Do something.
+        // result may be empty.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+### Search media ids by user metadata
+
+```java
+Map<String, String> filter = new HashMap<>();
+filter.put("meta.user.<key1>", "<value1>");
+filter.put("meta.user.<key2>", "<value2>");
+mstorage.search("<media_id>", filter, new CompletionHandler<MediaList>() {
+    @Override
+    public void onCompleted(MediaList mediaList) {
```

Comment:
According to [REST API](https://github.com/ricohapi/media-storage-rest/blob/master/media.md#post-mediasearch), we won't have "paging", so we do not need to use "MediaList" object. We can replace it with List<MediaIndex>.